mysql_sync/mysql_syncer.py

```diff
@@ -0,0 +1,153 @@
+
+
+def parse_file(base, cell_value):
+    if not cell_value:
+        return []
+    temp_file_name = uuid4().hex
+    file_info = base.upload_bytes_file(temp_file_name, cell_value)
+    return [file_info]
+
+
+def parse_multiple_select(cell_value):
+    if not cell_value:
+        return []
+    cell_value = str(cell_value)
+    values = cell_value.split(',')
+    return [value.strip() for value in values]
+
+
+def parse_long_text(cell_value):
+    if not cell_value:
+        return {}
+    # copy from dtable-web/frontend/src/components-form/utils/markdown-utils.js
+    HREF_REG = r'\[.+\]\(\S+\)|<img src=\S+.+\/>|!\[\]\(\S+\)|<\S+>'
+    LINK_REG_1 = r'^\[.+\]\((\S+)\)'
+    LINK_REG_2 = r'^<(\S+)>$'
+    IMAGE_REG_1 = r'^<img src="(\S+)" .+\/>'
+    IMAGE_REG_2 = r'^!\[\]\((\S+)\)'
+
+    cell_value = str(cell_value)
+    checked_count = cell_value.count('[x]')
+    unchecked_count = cell_value.count('[ ]')
+    total = checked_count + unchecked_count
+
+    href_reg = re.compile(HREF_REG)
+    preview = href_reg.sub(' ', cell_value)
+    preview = preview[:20].replace('\n', ' ')
+
+    images = []
+    links = []
+    href_list = href_reg.findall(cell_value)
+    for href in href_list:
+        if re.search(LINK_REG_1, href):
+            links.append(re.search(LINK_REG_1, href).group(1))
+        elif re.search(LINK_REG_2, href):
+            links.append(re.search(LINK_REG_2, href).group(1))
+        elif re.search(IMAGE_REG_1, href):
+            images.append(re.search(IMAGE_REG_1, href).group(1))
+        elif re.search(IMAGE_REG_2, href):
+            images.append(re.search(IMAGE_REG_2, href).group(1))
+
+    return {
+        'text': cell_value,
+        'preview': preview,
+        'checklist': {'completed': checked_count, 'total': total},
+        'images': images,
+        'links': links,
+    }
+
+
+class MysqlSync(object):
+    def __init__(self):
+        self.conn = pymysql.connect(user=settings.MYSQL_USER, password=settings.MYSQL_PASSWORD,
+                                    database=settings.DB_NAME, host=settings.MYSQL_HOST, charset=settings.CHARSET)
+        self.cursor = self.conn.cursor(cursor=pymysql.cursors.DictCursor)
+        self.base = Base(settings.BASE_API_TOKEN, settings.DTABLE_WEB_SERVICE_URL)
+        self.mode = settings.MODE
+        self.mysql_table_name = settings.MYSQL_TABLE_NAME
+        self.base_table_name = settings.BASE_TABLE_NAME
+        self.field_info = None
+
+    def now(self):
+        return str(datetime.now())
+
+    def base_auth(self):
+        self.base.auth()
+
+    def get_mysql_field_info(self):
+        sql = "DESC " + self.mysql_table_name
+        self.cursor.execute(sql)
+        field_info = self.cursor.fetchall()
+        self.field_info = {field['Field']: field['Type'].split('(')[0].upper() for field in field_info}
+
+    def get_mysql_data(self):
+        sql = "SELECT " + ','.join(self.field_info.keys()) + " FROM " + self.mysql_table_name
```

Comment:
','.join(self.field_info.keys()) -> ','.join(["`%s`" % key for key in self.field_info.keys()])